api/slack_test.go

```diff
@@ -0,0 +1,72 @@
+package api
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/slack-go/slack"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+type MockSlackClient struct {
+	mock.Mock
+}
+
+func (m *MockSlackClient) PostMessage(channelID string, options ...slack.MsgOption) (string, string, error) {
+	args := m.Called(channelID, options)
+	return args.String(0), args.String(1), args.Error(2)
+}
+
+func TestSendSlackMessages(t *testing.T) {
+	// Create a mock Slack client
+	mockClient := new(MockSlackClient)
+
+	// Set up test messages
+	messages := map[string]string{
+		"channel1": "message1",
+		"channel2": "message2",
+	}
+
+	// Test case 1: Successful send
+	mockClient.On("PostMessage", "channel1", mock.Anything, mock.Anything).Return("", "", nil).Once()
+
+	// Test case 2: Error sending Slack message
+	mockClient.On("PostMessage", "channel2", mock.Anything, mock.Anything).Return("", "", fmt.Errorf("Failed to send Slack message")).Once()
```

Comment:
We should test success and failure separately, so that those tests can succeed/fail separately.